app/logic/courseManagement.py

```diff
@@ -45,6 +48,58 @@ def approvedCourses(termId):
 
     return approvedCourses
 
+
+def importedCourses(termId):
+    '''
+    Queries the database to get all the necessary information for 
+    imported courses.
+    '''
+    importedCourses = (Course.select(Course, Term, CourseStatus, fn.GROUP_CONCAT(" " ,User.firstName, " ", User.lastName).alias('instructors'))
+                             .join(CourseInstructor, JOIN.LEFT_OUTER)
+                             .join(User, JOIN.LEFT_OUTER).switch(Course)
+                             .join(CourseStatus).switch(Course)
+                             .join(Term)
+                             .where(Term.id == termId, Course.status == CourseStatus.IMPORTED)
+                             .group_by(Course, Term, CourseStatus))
+
+    return importedCourses
+
+
+def editImportedCourses(courseData):
+    """
+        This function will take in courseData for the SLC proposal page and a dictionary
+        of instructors assigned to the imported course after that one is edited 
+        and update the information in the db.
+    """
+
+    with mainDB.atomic() as transaction:
+        try:
+            course = Course.get_by_id(courseData["courseId"])
+            
+            (Course.update(courseName=courseData["courseName"])
+                   .where(Course.id == course.id).execute()) # Update the data (Course Name) of the course in the database
+
+            (CourseParticipant.update(hoursEarned=courseData["hoursEarned"])
+                              .where(CourseParticipant.course_id == course.id).execute())
+            
+            instructorList = []
+            if 'instructor[]' in courseData:
+                instructorList = courseData.getlist('instructor[]') # Fetch the list of course instructors from CourseData
+            
+            if instructorList:
```

Comment:
A little nitpick but this will never execute if the previous conditional does not run, so I think you should combine these conditionals in some way.

Reply:
Fixed.